### PR TITLE
I18n routes with permalinks

### PR DESCRIPTION
**Pull request by martinratinaud**
In order to ease contribution, add support for `permalink` in mdx files of `content/pages` folder

For this, rework the whole I18n module so that's it's easier to user and abstract all complexity of this behavior, which is not supported by default on NextJs

This will result in
- a new `Link` component to be used instead of `next/link`
- a new `useTranslation` to be used instead of `next-translate/useTranslation`
- a new next.config plugin  to be used instead of `next-translate`
- a new `permalinks.json` file to handle translation of `src/pages` as it's not yet supported automatically
- a new `permalink` frontmatter entry in `content/pages` mdx files

All known page slugs have been translated and this should be tested on your local prior to approval

### CODE REVIEW

File: src/modules/I18n/README.md [hunk 1]
```diff
@@ -92,8 +111,43 @@ export default function TermsOfServicePage({ mdxContent }: WithI18nResult) {
...
 
-export const getStaticProps = withI18n({ load: 'mdx', folder: "static", filename: "terms-of-service" })();
+export const getStaticProps = withI18n({ folder: "static", filename: "terms-of-service" })();
+```
+
+### Translate slugs
+
+This is kind of a tricky thing to do with nextJs so this is a basic implementation that aims at being automatized better.
+
+1. Create a `permalinks.json` file in `modules/18n` for translating slugs of pages in `src/pages` folder
```
Comment (clementbiron) [inline src/modules/I18n/README.md:121]:
```suggestion
1. Create a `permalinks.json` file in `modules/i18n` for translating slugs of pages in `src/pages` folder
```

File: src/modules/I18n/README.md [hunk 2]
```diff
@@ -92,8 +111,43 @@ export default function TermsOfServicePage({ mdxContent }: WithI18nResult) {
...
+
+3. Replace all links with I18n Link
+
 ```
+import Link from 'next/link';
+# becomes
+import { Link } from 'modules/I18n';
+```
+
+4. Use `permalink` frontmatter value in `content/pages/*.mdx` files
```
Comment (clementbiron) [inline src/modules/I18n/README.md:150]: It would be better with an exemple.

**Review by clementbiron — CHANGES_REQUESTED** [review]
Why `/fr/souscrire` is redirecting to `/fr/subscribe` while `/fr/souscrire/elections-francaises` is working properly ?

**Comment by martinratinaud** [comment]
> Why `/fr/souscrire` is redirecting to `/fr/subscribe` while `/fr/souscrire/elections-francaises` is working properly ?

There is a custom `pages/subscribe.tsx` file. It's not a markdown one

**Review by clementbiron — APPROVED** [review]

**Pull request merged**